src/ORM/Marshaller.php

```diff
@@ -214,6 +214,9 @@ protected function _prepareDataAndOptions($data, $options)
      */
     protected function _marshalAssociation($assoc, $value, $options)
     {
+        if (!is_array($value)) {
+            return;
```

Comment:
There are a bunch of other invalid data states that are silently dropped already. Dropping invalid data feels more user friendly than blowing up on any issue.